src/menus/LabelMenus.cpp

```diff
@@ -182,18 +187,48 @@ void EditByLabel(AudacityProject &project,
    const bool notLocked = (!SyncLockState::Get(project).IsSyncLocked() &&
                            (tracks.Selected<PlayableTrack>()).empty());
 
-   //Apply action on tracks starting from
-   //labeled regions in the end. This is to correctly perform
-   //actions like 'Delete' which collapse the track area.
-   for (auto t : tracks) {
-      const bool playable = dynamic_cast<const PlayableTrack *>(t) != nullptr;
-      if (SyncLock::IsSyncLockSelected(t) || (notLocked && playable)) {
-         for (size_t i = regions.size(); i--;) {
-            const Region &region = regions.at(i);
-            action(*t, region.start, region.end);
-         }
-      }
+   std::vector<Track*> tracksToEdit;
+   for (auto t : tracks)
+   {
+      const bool playable = dynamic_cast<const PlayableTrack*>(t) != nullptr;
+      if (SyncLock::IsSyncLockSelected(t) || (notLocked && playable))
+         tracksToEdit.push_back(t);
    }
```

Comment:
I might write this without the vector:

`const auto range = tracks.Any<PlayableTrack>() + [&](const auto pTrack){ return SyncLock::IsSyncLockSelected(t) || (notLocked && playable); }`

Reply:
Then just use `range` twice in range-for statements

Reply:
IIUC, the `+` overload works like an "&&", and your proposal would yield another condition: `playable && (IsSynCLockSelected || notLocked)`.
I think what could work is
```
const auto range = tracks.Any<Track>() + [&](const auto pTrack) {
      return SyncLock::IsSyncLockSelected(t) ||
             (notLocked && dynamic_cast<const PlayableTrack*>(pTrack) != nullptr);
   };
```
but that's not so important.